test-integration/operator-tests/src/test/java/org/infinispan/operator/security/AuthTest.java

```diff
@@ -0,0 +1,72 @@
+package org.infinispan.operator.security;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.http.entity.ContentType;
+import org.assertj.core.api.Assertions;
+import org.infinispan.TestServer;
+import org.infinispan.crd.InfinispanContextProvider;
+import org.junit.jupiter.api.Test;
+
+import cz.xtf.client.Http;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShifts;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+
+abstract class AuthTest {
+   static final CustomResourceDefinitionContext infinispanContext = new InfinispanContextProvider().getContext();
+   static final OpenShift openShift = OpenShifts.master();
+
+   static final TestServer testServer = TestServer.get();
+   static final String testServerHost = testServer.host();
+
+   static String appName;
+   static String host;
+
+   static String user;
+   static String pass;
+
+   @Test
+   void restCorrectCredentials() throws IOException {
+      int putCode = Http.put("http://" + host + "/rest/testcache/correct").basicAuth(user, pass).data("credentials", ContentType.TEXT_PLAIN).execute().code();
+      int getCode = Http.get("http://" + host + "/rest/testcache/correct").basicAuth(user, pass).execute().code();
+
+      Assertions.assertThat(putCode).isEqualTo(204);
+      Assertions.assertThat(getCode).isEqualTo(200);
+   }
+
+   @Test
+   void restInvalidPassword() throws IOException {
+      Assertions.assertThat(Http.put("http://" + host + "/rest/testcache/invalid").basicAuth(user, "invalid").data("credentials", ContentType.TEXT_PLAIN).execute().code()).isEqualTo(401);
```

Comment:
Could you make these lines a bit narrower? Easier to review the PR without having to scroll ;)

Reply:
I'll see what I can do with that :)